src/features/import-map.js

```diff
@@ -4,46 +4,90 @@
  * <script type="systemjs-importmap">{}</script>
  * OR
  * <script type="systemjs-importmap" src=package.json></script>
- * 
- * Only supports loading the first import map
  */
 import { baseUrl as pageBaseUrl, parseImportMap, resolveImportMap } from '../common.js';
 import { systemJSPrototype } from '../system-core.js';
 
-var importMap, importMapPromise;
+var importMapPromise, fetchPromises = {};
+
+// Start downloading the import maps as soon as possible
 if (typeof document !== 'undefined') {
-  const scripts = document.getElementsByTagName('script');
-  for (let i = 0; i < scripts.length; i++) {
-    const script = scripts[i];
-    if (script.type !== 'systemjs-importmap')
-      continue;
-
-    if (!script.src) {
-      importMap = parseImportMap(JSON.parse(script.innerHTML), pageBaseUrl);
-    }
-    else {
-      importMapPromise = fetch(script.src)
+  const scriptsToDownload = document.querySelectorAll('script[type="systemjs-importmap"][src]');
+
+  for (let i = 0; i < scriptsToDownload.length; i++) {
+    const script = scriptsToDownload[i]
+    fetchPromises[script.src] = fetch(script.src)
       .then(function (res) {
         return res.json();
       })
-      .then(function (json) {
-        importMap = parseImportMap(json, script.src);
-      });
+      .then(function (data) {
+        return parseImportMap(data, script.src);
+      })
+  }
+}
+
+function retrieveImportMaps() {
+  if (typeof document !== 'undefined') {
+    // We must apply/merge the import maps in the order that they appear in the dom
+    const scripts = document.querySelectorAll('script[type="systemjs-importmap"]');
+    let pendingMaps = []
+
+    for (let i = 0; i < scripts.length; i++) {
+      const script = scripts[i];
+      if (!script.src) {
+        pendingMaps.push(parseImportMap(JSON.parse(script.innerHTML), pageBaseUrl));
+      }
+      else if (fetchPromises[script.src]) {
+        pendingMaps.push(fetchPromises[script.src])
+      } else {
+        pendingMaps.push(
+          fetch(script.src)
+          .then(function (res) {
+            return res.json();
+          })
+          .then(function (data) {
+            return parseImportMap(data, script.src);
+          })
+        );
+      }
     }
-    break;
+
+    // free the memory for this object -- we don't need it anymore after the first System.import happens
+    fetchPromises = null
+
+    const initialImportMap = { imports: {}, scopes: {} };
+
+    return Promise.all(pendingMaps).then(function(importMaps) {
+      return importMaps.reduce(function(finalMap, importMap) {
+        return mergeImportMap(finalMap, importMap);
+      }, initialImportMap);
+    });
```

Comment:
I would remove this entire section entirely for the single main `importMapsPromise` population method above.